timefold-solver/src/test/java/com/baeldung/timefoldsolver/ShiftScheduleConstraintProviderTest.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.timefoldsolver;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+import ai.timefold.solver.test.api.score.stream.ConstraintVerifier;
```

Comment:
Could you explain a bit, or provide a documentation link to the purpose of the ai.timefold.solver.test package helpers?

What does the typical intented development process look like for someone writing a solver with corresponding tests?

Reply:
Here's the documentation: https://timefold.ai/docs/timefold-solver/latest/constraints-and-score/score-calculation#constraintStreamsTestingIsolatedConstraints

Would you like that in this PR somehow? Or in the article?

Reply:
Changed article text to

The test-scoped timefold-solver-test dependency contains ConstraintVerifier, a helper to test each constraint in isolation. This improves maintenance: we can refactor a single constraint without breaking tests of other constraints.

Reply:
>Would you like that in this PR somehow? Or in the article?

Sure let's leave a link in the article if you feel it makes sense.

> The test-scoped timefold-solver-test dependency contains ConstraintVerifier, a helper to test each constraint in isolation. This improves maintenance: we can refactor a single constraint without breaking tests of other constraints.

I feel like this is a pretty useful feature to point out from a design perspective. Constraint design seems like the a critical part of the configuration, and I'm glad to see there's a supported way to help assist that design process by testing in isolation, good stuff!

Reply:
Thanks, I am happy to see appreciation and understanding of the grander design :) Article adjusted.

I 've added a general link to the documention at the bottom of the article:
  For more information, see [the Timefold Solver documentation](https://timefold.ai/docs/timefold-solver/latest/).
That link will never go stale.